includes/class-newspack-popups-inserter.php

```diff
@@ -100,6 +125,22 @@ public static function retrieve_popup() {
 			'posts_per_page' => 1,
 		];
 
+		$category_ids = array_map(
+			function( $cagegory ) {
+				return $cagegory->term_id;
+			},
+			$categories
+		);
+		if ( count( $category_ids ) ) {
+			$args['category__in'] = $category_ids;
+		} else {
+			$args['category__not_in'] = get_terms(
+				'category',
+				[
+					'fields' => 'ids',
+				]
+			);
```

Comment:
See my other comment about how to replace the `category__not_in` query.

Reply:
Change here: https://github.com/Automattic/newspack-popups/pull/15/commits/a73e69d1bee5f388e862020c0409d867a66b416a